doc/ip-interface/loopback-action/ip-interface-loopback-action-design.md

```diff
@@ -0,0 +1,318 @@
+
+### 2.3.2 Loopback action set interface Ethernet
+Loopback action set on interface Ethernet flow is the same as interface vlan flow besides config DB and app DB keys being used. Used keys are defined in [2.2.1 Config DB](#221-config-db).
+
+### 2.3.3 Loopback action set interface port-channel
+Loopback action set on interface port-channel flow is the same as interface vlan flow besides config DB and app DB keys being used. Used keys are defined in [2.2.1 Config DB](#221-config-db).
+
+### 2.3.4 IP interface creation
+If loopback action was not configured by the user, default loopback action is derived from SAI behavioral model which is forward.
+
+## 2.4 SAI
+The following will be used for setting IP interface loopback action.
+
+SAI API:
+```
+sai_router_intfs_api->set_router_interface_attribute()
+sai_router_intfs_api->create_router_interface()
+```
+
+SAI attribute:
+###### Table 2: SAI attribute
+| Loopback action  | SAI attribute                                     | Attribute value            |
+|------------------|---------------------------------------------------|----------------------------|
+| drop             | SAI_ROUTER_INTERFACE_ATTR_LOOPBACK_PACKET_ACTION  | SAI_PACKET_ACTION_DROP     |
+| forward          | SAI_ROUTER_INTERFACE_ATTR_LOOPBACK_PACKET_ACTION  | SAI_PACKET_ACTION_FORWARD  |
+
+
+```
+typedef enum _sai_router_interface_attr_t
+{
+    /**
+     * @brief Packet action when a packet ingress and gets routed on the same RIF
+     *
+     * @type sai_packet_action_t
+     * @flags CREATE_AND_SET
+     * @default SAI_PACKET_ACTION_FORWARD
+     */
+    SAI_ROUTER_INTERFACE_ATTR_LOOPBACK_PACKET_ACTION,
+
+} sai_router_interface_attr_t;
+```
+
+## 2.5 Statistics
+Packets that will dropped due to loopback action will be counted in RX_ERR in IP interface statistics.
+```
+root@sonic:~# show interfaces counters rif
+      IFACE    RX_OK    RX_BPS    RX_PPS    RX_ERR    TX_OK    TX_BPS    TX_PPS    TX_ERR
+-----------  -------  --------  --------  --------  -------  --------  --------  --------
+Ethernet232        0  0.00 B/s    0.00/s         0        0  0.00 B/s    0.00/s         0
+```
+
+## 2.6 CLI
+### 2.6.1 Config command
+
+The following command will be added to configure IP interface loopback action:
+```
+config interface loopback-action <interface-name> <action>
+```
+action options: "drop", "forward"
+
+Examples:
+```
+config interface loopback-action Ethernet248 drop
+config interface loopback-action Vlan100 forward
+```
+
+### 2.6.2 Show command
+The following command will be added to show the configured loopback action.
+IP interfaces that user did not configure loopback action on them, will not be shown in the table.
+```
+show ip interfaces loopback-action
+```
+
+Example:
+```
+root@sonic:~# show ip interfaces loopback-action
+Interface     Action      
+------------  ----------  
+Ethernet232   drop     
+Vlan100       forward     
+```
```

Comment:
why not adding that to the interface status command?
and in the case nothing is set it will be N/A

Reply:
I suggested to add it to "show ip interfaces". Marian insisted on a new command.